### PR TITLE
[Design2-107] Dropdowns update - minor styles and stories updates (for consistenncy)

### PR DESCRIPTION
**Pull request by levadadenys**
## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

### [Design2-107] Dropdowns update - minor styles and stories updates (for consistenncy)

Initially this PR was adding `readOnly` state to `dropdowns`. However, upon further discussion we decided not to add it and leave just `disabled` state as we had initially.  (see pr comments , and [this slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1713795342004929)).

Upon adding `readOnly` state I've found some minor bugs with disabled state customDropdown white color, and missing documentation for `checkboxDropdown` in storybook and some other minor stories related things worth improving. Removed `readOnly` state from this PR, but left these updates/fixes here. Here's the full list:

**CheckboxDropdown**:
* minor styles updates for consistency (for disabled state)
* fix and update documentation
* added `className` prop

**IconDropdown**:
* minor styles and stories updates for consistency (for disabled state)
* added `className` prop

**SimpleDropdown**:
* minor story update

## Links
- jira ticket: [DESIGN2-107](https://codedotorg.atlassian.net/browse/DESIGN2-107)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by bencodeorg** [comment]
What is the distinction between `readOnly` and `disabled`?

**Comment by bencodeorg** [comment]
ah nvm, I see more detail in the linked Slack thread. This feels likely to introduce confusion to me...

**Comment by levadadenys** [comment]
> ah nvm, I see more detail in the linked Slack thread. This feels likely to introduce confusion to me...

@bencodeorg readOnly is used in some forms where there's some value already selected and being passed into the form/submit. (that's why it's styled like regular dropdown, except it can't be opened). Disabled dropdown is used when dropdown is disabled and we don't use selected value(?)/dropdown itself don't affect anything (that's why it's styled as disabled(greyed out completely))

**Comment by bencodeorg** [comment]
Got it, reading up more on `readonly` here more broadly:

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly

image

Since this PR doesn't make any functional changes relating to form submissions using these select elements (just styling), I don't understand how it will be helpful for a user to see a visual difference between a disabled and readonly select when they're functionally the same thing?

I also wouldn't be surprised if developers don't pick up the distinction between 'disabled' and 'readonly' and start using them interchangeably, since they both set the `disabled` prop, and we end up with varied styling across the site.

**Comment by levadadenys** [comment]
> Got it, reading up more on `readonly` here more broadly:
> 
> https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly
> 
> image
> 
> Since this PR doesn't make any functional changes relating to form submissions using these select elements (just styling), I don't understand how it will be helpful for a user to see a visual difference between a disabled and readonly select when they're functionally the same thing?
> 
> I also wouldn't be surprised if developers don't pick up the distinction between 'disabled' and 'readonly' and start using them interchangeably, since they both set the `disabled` prop), and we end up with varied styling across the site

Agree. From my perspective it comes down to Figma designs also. Will discuss it with design team, maybe we'll be able to came up with just one disabled styles, but not sure yet.

**Comment by levadadenys** [comment]
> upon further discussion we decided not to add it and leave just `disabled` state as we had initially. (see pr comments , and [this slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1713795342004929)).

Upon further discussion we decided not to add it and leave just disabled state as we had initially. (see pr comments , and [this slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1713795342004929)).

**Comment by levadadenys** [comment]
> Agree. From my perspective it comes down to Figma designs also. Will discuss it with design team, maybe we'll be able to came up with just one disabled styles, but not sure yet.

Upon further discussion we decided not to add it and leave just disabled state as we had initially. (see pr comments , and [this slack thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1713795342004929)).

File: apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.story.tsx [hunk 1]
```diff
@@ -7,16 +7,14 @@ import CheckboxDropdown, {CheckboxDropdownProps} from './index';
 
 export default {
   title: 'DesignSystem/Dropdown/Checkbox Dropdown', // eslint-disable-line storybook/no-title-property-in-meta
-  component: CheckboxDropdown,
+  component: CheckboxDropdown.type,
```
Comment (bencodeorg) [inline apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.story.tsx:10]: For my understanding, what does this do?

File: apps/test/unit/componentLibrary/CheckboxDropdownTest.jsx [hunk 2]
```diff
@@ -91,7 +91,6 @@ describe('Design System - Checkbox Dropdown Component', () => {
     rerender(<DropdownToRender />);
 
     expect(spyOnChange).to.have.been.calledOnce;
-    expect(option1.checked).to.be.true;
```
Comment (bencodeorg) [inline apps/test/unit/componentLibrary/CheckboxDropdownTest.jsx:94]: Why don't we expect these to be checked any more?

**Review by bencodeorg — APPROVED** [review]

File: apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.story.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/dropdown/checkboxDropdown/CheckboxDropdown.story.tsx:10]: Oh yeah, that's some 'storybook magic' to enable storybook documentation generation when component used in the story is wrapped up in `React.memo` function. And the reason I've added `React.memo` is that documentation in storybook was not fetching from the code at all for some reason before

File: apps/test/unit/componentLibrary/CheckboxDropdownTest.jsx (hunk 2 above)
Comment (levadadenys) [reply apps/test/unit/componentLibrary/CheckboxDropdownTest.jsx:94]: This is caused by using `React.memo` on the other hand. For some reason - our RTL + jest combination of testing is not rerndering this component correctly, however values do change in the real code. So I implemented this test to work same way `IconDropdownTest` is working now, just checking if selected values did indeed change when selected

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**